Library/AzureVirtualMachine.ts

```diff
@@ -0,0 +1,61 @@
+
+import Config = require("./Config");
+import Util = require("./Util");
+import AutoCollectHttpDependencies = require("../AutoCollection/HttpDependencies");
+
+const AIMS_URI = "http://169.254.169.254/metadata/instance/compute";
+const AIMS_API_VERSION = "api-version=2017-12-01";
+const AIMS_FORMAT = "format=json";
+
+class AzureVirtualMachine {
+
+    public isVM: boolean;
+    public id: string;
+    public subscriptionId: string;
+    public osType: string;
+
+    constructor(config: Config) {
+        this.isVM = false;
+        this.id = "";
+        this.subscriptionId = "";
+        this.osType = "";
+        this.initialize(config);
+    }
+
+    private async initialize(config: Config) {
+        await this._getAzureComputeMetadata(config);
+    }
+
+    private async _getAzureComputeMetadata(config: Config) {
+        const metadataRequestUrl = `${AIMS_URI}?${AIMS_API_VERSION}&${AIMS_FORMAT}`;
+        const requestOptions = {
+            method: 'GET',
+            [AutoCollectHttpDependencies.disableCollectionRequestOption]: true,
+            headers: {
+                "Metadata": "True",
+            }
+        };
+
+        const req = Util.makeRequest(config, metadataRequestUrl, requestOptions, (res) => {
+            if (res.statusCode === 200) {
```

Comment:
you might want to log when it's not reachable? non-vm will lead to socketexception.  in that case, it's a good indicator that it's not from a vm.. you can shutdown the scheduler if applicable.

Reply:
That must be logged already in line 68 for latest commit